ext/mkrf_conf.rb

```diff
@@ -0,0 +1,63 @@
+# Inspired by:
+# https://github.com/ruby-debug/ruby-debug-ide/blob/master/ext/mkrf_conf.rb
+
+# This file needs to be named mkrf_conf.rb
+# so that rubygems will recognize it as a ruby extension
+# file and not think it is a C extension file
+
+require 'rubygems/specification'
+require 'rubygems/dependency'
+require 'rubygems/dependency_installer'
+
+# Load up the rubygems dependency installer to install the dependencies
+# we need based on the platform we are running under.
+installer = Gem::DependencyInstaller.new
+deps = []
+begin
+  # Try to detect Cisco NX-OS and IOS XR environments
+  os = nil
+  if File.exist?('/etc/os-release')
+    cisco_release_file = nil
+    File.foreach('/etc/os-release') do |line|
+      next unless line[/^CISCO_RELEASE_INFO=/]
+      cisco_release_file = line[/^CISCO_RELEASE_INFO=(.*)$/, 1]
+      break
+    end
+    unless cisco_release_file.nil?
+      File.foreach(cisco_release_file) do |line|
+        next unless line[/^ID=/]
+        os = line[/^ID=(.*)$/, 1]
+        break
+      end
+    end
+  end
+  puts "Detected client OS as '#{os}'" unless os.nil?
+
+  # IOS XR doesn't need net_http_unix
+  os == 'ios_xr' || deps << Gem::Dependency.new('net_http_unix',
+                                                '~> 0.2', '>= 0.2.1')
+  # NX-OS doesn't need gRPC
+  os == 'nexus' || deps << Gem::Dependency.new('grpc', '~> 0.11')
+
+  deps.each do |dep|
+    installed = dep.matching_specs
+    if installed.empty?
+      puts "Installing #{dep}"
+      installed = installer.install dep
+      fail installer.errors[0] unless installer.errors.empty?
```

Comment:
For some reason `Gem::DependencyInstaller.install` captures exceptions during installation and stores them in a local array rather than raising them to the caller. I need to fail if the installation failed, and since I can't throw multiple exceptions at once, I just throw the first exception in the array to indicate failure.